packages/itwin/ec3-widget/src/components/TemplateMenu.tsx

```diff
@@ -95,170 +56,144 @@ export const TemplateMenu = ({ template, onSaveSuccess, onClickCancel }: Templat
           labels: configuration.labels,
         };
         setChildTemplate(childConfig);
-      }
-
-      try {
-        const accessToken = await getAccessToken();
-        const data = await reportsClient.getReports(accessToken, projectId);
-        if (data) {
-          const fetchedReports = data.sort((a, b) => a.displayName?.localeCompare(b.displayName ?? "") ?? 0);
-          setReports(fetchedReports);
-          setIsLoading(false);
+      } else {
+        try {
+          const accessToken = await getAccessToken();
+          const data = await reportsClient.getReports(accessToken, projectId);
+          if (data && data.length > 0) {
+            // take report coming from consumer or take the first available report
```

Comment:
"Take first available report" sounds sketchy. Users might have multiple reports for different purposes. If defaultReport is not provided or not found, user should be allowed to pick a report or should be given instructions on how to create it.

Reply:
added ability for the user to select report, if a default report is not sent by the consumer